### PR TITLE
First integration tests running on Python 3

### PR DESCRIPTION
**Pull request by itamarst**
Fixes https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3703

### CODE REVIEW

**Comment by exarkun** [comment]
I can't find where GitHub Actions is running the ported integration test module.  Any hints?

**Comment by itamarst** [comment]
It's running in same runners doing unit tests.

**Comment by exarkun** [comment]
That's what the tox.ini changes led me to believe but I still can't find them.

**Comment by itamarst** [comment]
Perhaps it's not running... let me look at logs.

**Comment by itamarst** [comment]
Yep, not running. So I guess I'll fix that next :)

**Comment by itamarst** [comment]
OK, it's working now, specific on the Ubuntu 3.6 runner.

**Comment by itamarst** [comment]
The coverage decrease suggests I screwed something up re coverage?

**Comment by sajith** [comment]
About coverage decrease: is it that unit test coverage data gets thrown away or overwritten?  I am not entirely sure how coverage works, but, perhaps `coverage combine` should be done only once?  

Maybe we could add `3.6` to `jobs.integration.strategy.matrix.python-version`?  Also, do we want/need coverage report from integration tests yet?  As far as I know, we have not really looked at coverage of integration tests before, and we do not have much of a plan to act upon that information.

**Review by sajith — APPROVED** [review]
Looks good to me, except that it looks like running integration tests in a separate job would be useful.  That might help us in avoiding confusing coverage reports.

File: src/allmydata/util/_python3.py [hunk 1]
```diff
@@ -22,6 +22,15 @@
 if PY2:
     from future.builtins import filter, map, zip, ascii, chr, hex, input, next, oct, open, pow, round, super, bytes, dict, list, object, range, str, max, min  # noqa: F401
 
+
+# Every time a module is added here, also add it to tox.ini environment
+# integrations3. Bit of duplication, but it's only a handful of files so quite
+# temporary.
+PORTED_INTEGRATION_TESTS = [
+    "integration.test_servers_of_happiness",
+]
```
[the remark this answers is not in the document]
Comment (itamarst) [reply src/allmydata/util/_python3.py:31]: No it's just for tracking purposes, it's not used anywhere. By the 3rd PR all of them are ported and it's not necessary any more and can be deleted.

**Pull request merged**